CHANGELOG.md

```diff
@@ -0,0 +1,50 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
+and follows a [Backwards Compatability Policy]
+(https://docs.solana.com/developing/backwards-compatibility)
+
+## Unreleased
+### [1.17.0]
+* Changes
+  * Added a changelog!
+* Upgrade Notes
+
+## Released
+### 2023-05-31 - [1.16.0](https://github.com/solana-labs/solana/releases/tag/v1.16.0)
+
+## Adding to this Changelog
+### Audience
+* Entries in this log are intended to be easily understood by contributors,
+consensus validator operators, rpc operators, and dapp developers.
+
+### Noteworthy
+* A change is noteworthy if it:
+  * Adds a feature gate, or
+  * Implements a SIMD, or
+  * Modifies a public API, or
+  * Changes normal validator / rpc run configurations, or
+  * Changes command line arguments, or
+  * Fixes a bug that has received public attention, or
+  * Significantly improves performance, or
+  * Is authored by an external contributor.
```

Comment:
i was hoping we could filter the refspec compare query by author, like the branch commits list can be. then we could just bash up a list of links for each external contributor's commits in the version rather than call out each individual contribution. doesn't look like that's possible today though. hopefully this doesn't turn into a dumb scoreboard.

Reply:
Agreed that calling out each individual contribution could be a lot.

Throwing out an idea I saw elsewhere - [rust-rocksdb has a section specifically for first time contributors](https://github.com/rust-rocksdb/rust-rocksdb/releases). Could potentially do something similar where first time external contributors always get a shout out, and then subsequent changes are subject to the above criteria.

I'm not sure how that list is generated tho, and `rust-rocksdb` has a much lower commit volume than this repo so it is possible that the list is created manually. Not sure we'd want to take that on